### PR TITLE
Npm fix

### LINKED ISSUE #1418: Error when running master locally
`npm i` and then `npm start` -->
```
ERROR in ./node_modules/ethereum-cryptography/src/hash-utils.ts 4:17
Module parse failed: Unexpected token (4:17)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
| 
| export function createHashFunction(
>   hashConstructor: () => Hash
| ): (msg: Buffer) => Buffer {
|   return msg => {
 @ ./node_modules/ethereum-cryptography/src/keccak.ts 1:0-50 5:25-43 9:25-43 13:25-43 17:25-43
 @ ./node_modules/ethereum-cryptography/keccak.js
 @ ./node_modules/ethereumjs-util/dist/index.js
 @ ./node_modules/web3-provider-engine/subproviders/hooked-wallet.js
 @ ./node_modules/@walletconnect/web3-provider/dist/cjs/index.js
 @ ./src/utils/autoconnect.ts
 @ ./src/utils/index.ts
 @ ./src/App.tsx
 @ ./src/index.tsx

ERROR in ./node_modules/ethereum-cryptography/src/secp256k1.ts 6:40
Module parse failed: Unexpected token (6:40)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
| const SECP256K1_PRIVATE_KEY_SIZE = 32;
| 
> export async function createPrivateKey(): Promise<Uint8Array> {
|   while (true) {
|     const pk = await getRandomBytes(SECP256K1_PRIVATE_KEY_SIZE);
 @ ./node_modules/ethereum-cryptography/secp256k1.js 42:18-38 74:9-29
 @ ./node_modules/ethereumjs-util/dist/secp256k1-adapter.js
 @ ./node_modules/ethereumjs-util/dist/index.js
 @ ./node_modules/web3-provider-engine/subproviders/hooked-wallet.js
 @ ./node_modules/@walletconnect/web3-provider/dist/cjs/index.js
 @ ./src/utils/autoconnect.ts
 @ ./src/utils/index.ts
 @ ./src/App.tsx
 @ ./src/index.tsx
```

### PR DESCRIPTION
**Pull request by Velenir**
- changes what dependency versions `@walletconnect/web3-provider` uses, had to edit `package-lock.json` because semver inside  `@walletconnect/web3-provider`  causes `@walletconnect/core` and other dependencies to be vastly different (`1.0.0-rc.0 vs  1.2.2`). I can't stress enough the need to update all `@walletconnect/*` deps with #1245 
- 
- fixes resolution for sub-dependencies from `src` folder local to them (`ethereum-cryptography/src/hash-utils.ts` issue)
- 
- workaround for `console-feed` regenerator runtime issue, that arose with deps updates
- 
- `package-lock.json` is now comitted to have controlled deps for npm
-
- linter and TS fixes that arose with deps updates

I experience a strange issue that `eslint` lints differently when deps are installed with npm and when installed with yarn

Fixes #1418

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr1419--dexreact.review.gnosisdev.com

File: .gitignore [hunk 1]
```diff
@@ -7,7 +7,6 @@ dist
 *.log
 coverage
 .env
-package-lock.json
```
Comment (anxolin) [inline .gitignore:10]: why are we bringing back this file?

File: jest.config.js [hunk 2]
```diff
@@ -9,7 +10,7 @@ module.exports = {
     '^.+\\.tsx?$': 'ts-jest',
   },
   testRegex: '(/__tests__/.*|(\\.|/)(test|spec))\\.tsx?$',
-  moduleDirectories: ['node_modules', 'src'],
```
Comment (anxolin) [inline jest.config.js:12]: How this is necessary? https://jestjs.io/docs/en/configuration#moduledirectories-arraystring

File: package.json [hunk 3]
```diff
@@ -13,7 +13,7 @@
     "mock": "MOCK=true yarn start",
     "build": "rimraf ./dist && NODE_ENV=production webpack --mode production",
     "lint": "eslint src --ext .ts,.tsx,.js",
-    "lint:fix": "eslint src --fix . --ext .ts,.tsx,.js",
```
Comment (anxolin) [inline package.json:16]: I overlooked that we don't lint with that the tests. Is this intended?

File: webpack.config.js [hunk 4]
```diff
@@ -111,8 +111,9 @@ module.exports = ({ stats = false } = {}) => ({
     alias: {
       'react-dom': '@hot-loader/react-dom',
       'bn.js': path.resolve(__dirname, 'node_modules/bn.js'),
+      'react-inspector': path.resolve(__dirname, 'node_modules/react-inspector'),
     },
-    modules: ['custom', 'src', 'node_modules'],
+    modules: [path.resolve(__dirname, 'custom'), path.resolve(__dirname, 'src'), 'node_modules'],
```
Comment (anxolin) [inline webpack.config.js:116]: That's strange too

**Review by anxolin — COMMENTED** [review]
Great Dima! it works now for me!
I don't get 100% why this fixes it. My understanding is NPM was resolving the dependencies slightly differently, so it was bringing some TS file as a transitive dependency instead of it's transpiled file.
Did you understand the reason why that was happening?

I understand this PR does several things that are not needed to fix the issue and you plan to narrow down and cleanup. 

I see there's formatting that is totally unrelated. If we need to reformat the files, please do it in a different PR in isolation pls. We don't want important changes together with reformatting.

File: package.json (hunk 3 above)
Comment (Velenir) [reply package.json:16]: Don't know if intended, don't remember, the problem was the dot after `--fix`

**Comment by Velenir** [comment]
> I understand this PR does several things that are not needed to fix the issue and you plan to narrow down and cleanup.
> 
> I see there's formatting that is totally unrelated. If we need to reformat the files, please do it in a different PR in isolation pls. We don't want important changes together with reformatting.

All changes in this PR are related as wrote `that arose with deps updates`
TS problems and lint problems didn't allow to commit because of precommit hook  with `npm run verify`. Would you have preferred a `--no-verify` commit? I wouldn't. An even with `--no-verify` I would expect travis to break

**Comment by Velenir** [comment]
> Did you understand the reason why that was happening?

Problem was in `ethereum-cryptography` lib which is used in `ethereumjs-util>=6.2.1`
Yarn would install [6.2.0](https://unpkg.com/browse/ethereumjs-util@6.2.0/package.json#LC90), npm -- [6.2.1](https://unpkg.com/browse/ethereumjs-util@6.2.1/package.json#LC94), very close versions within semver of whatever depends on them

File: .gitignore (hunk 1 above)
Comment (Velenir) [reply .gitignore:10]: > package-lock.json is now comitted to have controlled deps for npm

same reason we have `yarn.lock`

File: jest.config.js (hunk 2 above)
Comment (Velenir) [reply jest.config.js:12]: Same reason it's changed in webpack config

From the doc you linked:
> An array of directory names to be searched recursively up from the requiring module's location

guess what happens when there's src in `node_modules/some_dep/src`

File: webpack.config.js (hunk 4 above)
Comment (Velenir) [reply webpack.config.js:116]: That's actually the recommended way now that I check [resolve.modules](https://webpack.js.org/configuration/resolve/#resolvemodules:~:text=path.resolve(__dirname%2C%20'src')%2C)

File: .gitignore (hunk 1 above)
Comment (anxolin) [reply .gitignore:10]: We use yarn.lock for predictable building in travis, I think keeping the 2 of them is confusing...

File: jest.config.js (hunk 2 above)
Comment (anxolin) [reply jest.config.js:12]: Really? Why does it have precedence a node_modules than our own `src`

File: package.json (hunk 3 above)
Comment (anxolin) [reply package.json:16]: Yep, I understand. Just asking, cause this would allow to commit code with formatting issues in the tests

**Review by anxolin — APPROVED** [review]

File: jest.config.js (hunk 2 above)
Comment (Velenir) [reply jest.config.js:12]: don't know, who wrote it, must have had a reason

**Pull request merged**